pipelines/pulpcore/03-tests.yml

```diff
@@ -1,10 +1,18 @@
 ---
 - name: run tests
-  hosts:
-    - "{{ forklift_server_name }}"
   become: True
+  hosts:
+   - "{{ forklift_name }}"
   vars_files:
-    - ../vars/install_base.yml
-    - ../vars/forklift_{{ pipeline_type }}.yml
-  roles:
-    - pulp.pulp_installer.pulp_health_check
+    - ../vars/forklift_pulpcore.yml
+  tasks:
+    - name: Run acceptance tests
+      command: bundle exec rake beaker
+      args:
+        chdir: /src/puppet-pulpcore
+      environment:
+        BEAKER_HYPERVISOR: "docker"
```

Comment:
I don't particularly like that we spin up a centos box just to run a centos container inside, especially as our primary way of deployment is not a container.

Shouldn't beaker default to using the machine it's running on when we do not set BEAKER_HYPERVIRSOR?

Reply:
I would argue it's not *just* to run a container inside -- we also install dependencies needed for the puppet module to perform testing: https://github.com/theforeman/forklift/pull/1729/files#diff-dec5007e2a651ace599f5154ceb5187aa75fb8373133bac84a45fd15840fbe9dR19-R41

Especially for our CI, that means we have a clean environment every time (yes, I know CentOS CI **starts** with a clean box and thus we could shorten this). Locally, this also means we have a clean environment every time. And, it's consistent.

Reply:
Fair.

I was also thinking about `BEAKER_HYPERVISOR: vagrant` which would fit nicely if we wouldn't run this all from inside a vagrant box already.

It's hypervisors all the way down.

Reply:
We could possibly do container-inception as well, however, I was not trying to re-invent the wheel here. Rather my goal was to switch away from the Ansible based Pulp installer and to using our own test suite that we vet a working Pulp against.

Reply:
Today we spin up very expensive bare metal boxes so we can use virtualization. If we don't, it's much cheaper for CentOS CI if we request a VM. So I'd say we either go with `BEAKER_HYPERVISOR=vagrant_libvirt` so we can also test SELinux (which we can't inside containers) or enhance our Duffy integration so it can request VMs via a parameter.

Today that's hardcoded here:
https://github.com/theforeman/jenkins-jobs/blob/d78f0f3b828444b109105fa5d27927be3f158c31/centos.org/ansible/provision_duffy.yml#L6

Perhaps also a good excuse to address https://github.com/theforeman/foreman-infra/issues/1828.